app/src/main/java/org/oppia/app/onboarding/OnboardingPagerAdapter.kt

```diff
@@ -0,0 +1,32 @@
+package org.oppia.app.onboarding
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.viewpager.widget.PagerAdapter
+import org.oppia.app.databinding.OnboardingSlideBinding
+
+/** Adapter to control the slide details in onboarding flow. */
+class OnboardingPagerAdapter(val context: Context) : PagerAdapter() {
+  override fun instantiateItem(container: ViewGroup, position: Int): Any {
+    val binding = OnboardingSlideBinding.inflate(LayoutInflater.from(context), container, false)
+    val onboardingSlideViewModel = OnboardingSlideViewModel(context, position)
+    binding.viewModel = onboardingSlideViewModel
+    container.addView(binding.root)
+    return binding.root
+  }
+
+  override fun getCount(): Int {
+    return TOTAL_NUMBER_OF_SLIDES
+  }
+
+  override fun isViewFromObject(view: View, `object`: Any): Boolean {
+    return view == `object`
```

Comment:
Can you explain this logic? I’m not sure I follow.

Reply:
The method `instantiateItem(ViewGroup, int)` returns Object for a particular view. PagerAdapter implementation is considering this Object as a key value when viewpager changes a page.

So, if we return the view itself from `instantiateItem(ViewGroup, int)`, then our key for that page becomes the view itself. We can check `return view == object;` from `isViewFromObject (View view, Object object)` which will always return true and our pages will display


@BenHenning Actually, this was a part of PagerAdapter extension and some google searches.

Main you can find the detail description here: https://stackoverflow.com/a/32709553